src/sentry/search/events/filter.py

```diff
@@ -961,10 +962,22 @@ def format_search_filter(term, params):
             for p in Project.objects.filter(id__in=params.get("project_id", []), slug__in=slugs)
         }
         missing = [slug for slug in slugs if slug not in projects]
-        if missing and term.operator in EQUALITY_OPERATORS:
-            raise InvalidSearchQuery(
-                f"Invalid query. Project(s) {oxfordize_list(missing)} do not exist or are not actively selected."
-            )
+        if missing:
+            if term.operator in EQUALITY_OPERATORS:
+                raise InvalidSearchQuery(
+                    f"Invalid query. Project(s) {oxfordize_list(missing)} do not exist or are not actively selected."
+                )
+            elif term.operator in INEQUALITY_OPERATORS:
+                # Set search filter to project name instead of id
+                term = SearchFilter(
+                    SearchKey("project.name"),
```

Comment:
not sure if this is allowed. it's supposed to convert to `project_id` and convert the name to the id as well, but if it doesn't exist that won't work

Reply:
I think searching on `project.name` results in a snuba query on `tags[project.name]` - not totally sure if SDKs or users send events with tag[project.name] values. In this case if the project slugs are missing and couldn't be resolved, could we just not add the filter to the condition?

Reply:
Actually, there's probably a flaw in that approach. 

case when len(missing) == len(slugs):
  # couldn't resolve any of the project slugs to a project so we don't need to filter down on project.

case when len(missing) < len(slugs):
  # some slugs are un-resolved, but thats ok just let the code down below filter by project_id


Hmm, what if we

Reply:
Yeah I was running into issues with the mixed case where some projects didn't exist/weren't selected and some were. I realized the problem was actually higher up the chain but regardless the filtering logic is a little restrictive down here.

Reply:
Going to cut another PR to clean up the code once this is merged